packages/@aws-cdk/aws-lambda-nodejs/lib/bundling.ts

```diff
@@ -67,13 +73,17 @@ export class Bundling implements cdk.BundlingOptions {
 
     Bundling.esbuildInstallation = Bundling.esbuildInstallation ?? EsbuildInstallation.detect();
 
-    this.projectRoot = path.dirname(props.depsLockFilePath);
+    this.projectRoot = props.projectRoot;
     this.relativeEntryPath = path.relative(this.projectRoot, path.resolve(props.entry));
 
     if (props.tsconfig) {
       this.relativeTsconfigPath = path.relative(this.projectRoot, path.resolve(props.tsconfig));
     }
 
+    if (props.depsLockFilePath) {
```

Comment:
This is always `true`, you can drop the `if` and then we're good.

(bonus: I think you can `throw` if `this.relativeDepsLockFilePath.includes('..')` which validates that `depsLockFilePath` is under `projectRoot`)

Reply:
Good catch. Updated both